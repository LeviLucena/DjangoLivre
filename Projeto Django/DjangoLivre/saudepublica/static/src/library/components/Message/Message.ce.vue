<template>
  <div v-if="active && !feedback" class="br-message" :class="stateClass" role="alert">
    <div v-if="showIcon" class="icon">
      <IconBase :icon-name="checkIconState(state)" class-name="fa-lg"></IconBase>
    </div>
    <div class="content">
      <!-- Se o usuario do componente nao passar a propriedade message indica que ele vai usar o slot -->
      <div v-if="!message">
        <strong v-if="title">{{ title }}</strong>
        <br v-if="title && !inline" />
        <slot></slot>
      </div>
      <div v-else>
        <span class="message-title mr-1">
          {{ title }}
        </span>
        <span class="message-body">
          {{ message }}
        </span>
      </div>
    </div>
    <div v-if="closable && !feedback" class="close">
      <button class="br-button circle small" type="button" aria-label="Fechar" @click="dismiss()">
        <IconBase icon-name="times"></IconBase>
      </button>
    </div>
  </div>
  <p v-if="feedback">
    <span class="feedback" role="alert" :class="stateClass">
      <IconBase v-if="showIcon" :icon-name="checkIconState(state)" />
      <div v-if="!message">
        <slot></slot>
      </div>
      <div v-else>
        {{ message }}
      </div>
    </span>
  </p>
</template>

<script>
import IconBase from "../IconBase/IconBase.ce.vue"
const Message = {
  name: "brMessage",
  components: {
    IconBase,
  },
  props: {
    /**
     * **[OPCIONAL]** Mensagem contextual, refere-se diretamente à ação do usuário sobre algum Componente. Por exemplo, a mensagem de validação de campo de texto.
     */
    feedback: {
      default: false,
      type: Boolean,
    },
    /**
     * **[OBRIGATÓRIO]** Define os estados. **state = "info | warning | danger |success"**
     */
    state: {
      default: "info",
      type: String,
      validator: function (value) {
        return ["info", "warning", "danger", "success"].includes(value)
      },
    },
    /**
     * **[OPCIONAL]** Por padrão, o título é posicionado no início, ACIMA da mensagem. Serve dar destaque à mensagem textual. Não disponível para mensagem do tipo feedback.
     */
    title: {
      type: String,
      default: "",
    },
    /**
     * **[OBRIGATÓRIO]** Define se a mensagem deve ser exibida.
     */
    message: {
      type: String,
      default: "",
    },
    /**
     * **[OPCIONAL]** Para que a propriedade "title" seja incluída no início, NA MESMA LINHA da mensagem.
     */
    inline: {
      default: false,
      type: Boolean,
    },
    /**
     * **[OPCIONAL]** Tem a função de ocultar o Componente Message da tela por meio do botão de fechar. Não disponível para mensagem do tipo feedback.
     */
    closable: {
      default: false,
      type: Boolean,
    },
    /**
     * **[OPCIONAL]** Tem a função de ocultar/mostrar o ícone da Message da tela.
     */
    showIcon: {
      default: false,
      type: Boolean,
    },
  },
  data() {
    return {
      active: true,
    }
  },
  computed: {
    stateClass() {
      return [
        {
          info: this.state === "info",
          warning: this.state === "warning",
          danger: this.state === "danger",
          success: this.state === "success",
        },
      ]
    },
  },
  methods: {
    dismiss() {
      this.active = false
    },
    checkIconState() {
      const iconMap = {
        success: "check-circle",
        warning: "exclamation-triangle",
        danger: "times-circle",
        info: "info-circle",
      }
      return iconMap[this.state] || "check-circle"
    },
  },
}

export default Message
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/message/_message";
@import "~@govbr-ds/core/src/components/button/_button";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
