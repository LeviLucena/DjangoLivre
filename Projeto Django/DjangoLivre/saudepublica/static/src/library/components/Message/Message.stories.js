import { kebabiseArgs } from '../../util/Utils.js'
import BrMessage from './Message.ce.vue'

const slotDefault = `Texto da mensagem.`

export default {
  title: 'Componentes/Message',
  component: BrMessage,
  argTypes: {
    state: {
      type: 'radio',
      options: ['info', 'warning', 'danger', 'success'],
      defaultValue: '',
    },
    title: {
      control: 'text',
      defaultValue: '',
    },
    inline: {
      control: 'boolean',
      defaultValue: false,
    },
    feedback: {
      control: 'boolean',
      defaultValue: false,
    },
    closable: {
      control: 'boolean',
      defaultValue: false,
    },
    showIcon: {
      control: 'boolean',
      defaultValue: false,
    },
    default: {
      control: 'text',
      description: '**[OBRIGATÓRIO]** Transmissão de informação de feedback do sistema.',
      table: {
        type: {
          summary: 'string | NodeList',
        },
        defaultValue: {
          summary: '',
        },
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-message v-bind="args">${args.default}</br-message>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  state: 'info',
  default: slotDefault,
}

export const feedback = Template.bind({})
feedback.args = {
  state: 'info',
  feedback: true,
  default: slotDefault,
}

export const closable = Template.bind({})
closable.args = {
  state: 'info',
  closable: true,
  default: slotDefault,
}

export const title = Template.bind({})
title.args = {
  state: 'info',
  title: 'Texto da mensagem com título.',
  default: slotDefault,
}

export const titleInline = Template.bind({})
titleInline.args = {
  state: 'info',
  title: 'Texto da mensagem com título inline.',
  inline: true,
  default: slotDefault,
}
