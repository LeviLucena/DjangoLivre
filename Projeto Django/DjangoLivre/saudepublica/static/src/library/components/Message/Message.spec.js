import { shallowMount } from '@vue/test-utils'
import Message from './Message.ce.vue'

describe('br-message', () => {
  test('checks if the message is feedback type', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        feedback: true,
      },
    })
    expect(wrapper.find('.feedback').exists()).toBe(true)
  })

  test('assign state danger and check if it is true', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        state: 'danger',
      },
    })
    expect(wrapper.find('.br-message.danger').exists()).toBe(true)
  })

  test('assign state info and check if it is true', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        state: 'info',
      },
    })
    expect(wrapper.find('.br-message.info').exists()).toBe(true)
  })

  test('assign state warning and check if it is true', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        state: 'warning',
      },
    })
    expect(wrapper.find('.br-message.warning').exists()).toBe(true)
  })

  test('assign state success and check if it is true', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        state: 'success',
      },
    })
    expect(wrapper.find('.br-message.success').exists()).toBe(true)
  })

  test('assign and checks if title property exists in a message', () => {
    const titulo = 'Titulo da mensagem'
    const wrapper = shallowMount(Message, {
      propsData: {
        title: titulo,
      },
    })
    expect(wrapper.text()).toMatch(titulo)
  })

  test('checks if title property do NOT exists in feedback', () => {
    const titulo = 'Titulo da mensagem'
    const wrapper = shallowMount(Message, {
      propsData: {
        feedback: true,
        title: titulo,
      },
    })
    expect(wrapper.find('strong').exists()).toBe(false)
  })

  test('assign inline property to title and check if it is true', () => {
    const titulo = 'Titulo da mensagem'
    const wrapper = shallowMount(Message, {
      propsData: {
        title: titulo,
        inline: true,
      },
    })
    expect(wrapper.find('p').exists()).toBe(false)
  })

  test('assign closable property to message and check if it is true', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        closable: true,
      },
    })
    expect(wrapper.find('.close').exists()).toBe(true)
  })

  test('assign closable property to feedback and check if it do NOT exists', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        feedback: true,
        closable: true,
      },
    })
    expect(wrapper.find('.close').exists()).toBe(false)
  })

  test('emit dismiss method on closable message', () => {
    const spyClose = jest.spyOn(Message.methods, 'dismiss')
    const wrapper = shallowMount(Message, {
      propsData: {
        closable: true,
      },
    })
    const btnClose = wrapper.find('.close button')
    btnClose.trigger('click')
    expect(spyClose).toHaveBeenCalled()
    spyClose.mockReset()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('message has the expected html structure - snapshot', () => {
    const titulo = 'Titulo da mensagem'
    const wrapper = shallowMount(Message, {
      propsData: {
        closable: true,
        title: titulo,
      },
      slots: {
        default: 'Texto da mensagem.',
      },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('feedback has the expected html structure - snapshot', () => {
    const wrapper = shallowMount(Message, {
      propsData: {
        feedback: true,
      },
      slots: {
        default: 'Texto da mensagem.',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
