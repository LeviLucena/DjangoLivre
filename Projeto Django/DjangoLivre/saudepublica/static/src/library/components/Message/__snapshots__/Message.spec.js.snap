// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`br-message emit dismiss method on closable message 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="br-message info"
    role="alert"
  >
    <!--v-if-->
    <div
      class="content"
    >
      <!-- Se o usuario do componente nao passar a propriedade message indica que ele vai usar o slot -->
      <div>
        <!--v-if-->
        <!--v-if-->
        
        
      </div>
    </div>
    <div
      class="close"
    >
      <button
        aria-label="Fechar"
        class="br-button circle small"
        type="button"
      >
        <icon-base-stub
          class=""
          familyname="fas"
          iconname="times"
        />
      </button>
    </div>
  </div>
  <!--v-if-->
  
</div>
`;

exports[`br-message feedback has the expected html structure - snapshot 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <p>
    <span
      class="feedback info"
      role="alert"
    >
      <!--v-if-->
      <div>
        
        Texto da mensagem.
        
      </div>
    </span>
  </p>
  
</div>
`;

exports[`br-message message has the expected html structure - snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="br-message info"
    role="alert"
  >
    <!--v-if-->
    <div
      class="content"
    >
      <!-- Se o usuario do componente nao passar a propriedade message indica que ele vai usar o slot -->
      <div>
        <strong>
          Titulo da mensagem
        </strong>
        <br />
        
        Texto da mensagem.
        
      </div>
    </div>
    <div
      class="close"
    >
      <button
        aria-label="Fechar"
        class="br-button circle small"
        type="button"
      >
        <icon-base-stub
          class=""
          familyname="fas"
          iconname="times"
        />
      </button>
    </div>
  </div>
  <!--v-if-->
  
</div>
`;
